akka-typed/src/main/scala/akka/typed/javadsl/BehaviorBuilder.scala

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.typed.javadsl
+
+import scala.annotation.tailrec
+import akka.japi.pf.FI
```

Comment:
Seems to be quite some overlap between `akka.japi.*` and `akka.japi.function.*` too - the latter is preferred? let's perhaps add @deprecated notices in another PR?